diff --git a/sysdrv/Makefile b/sysdrv/Makefile
index 3ffe067e3..153830d81 100644
--- a/sysdrv/Makefile
+++ b/sysdrv/Makefile
@@ -721,9 +721,9 @@ drv: prepare
 		O=$(SYSDRV_KERNEL_OBJS_OUTPUT_DIR) \
 		modules_install INSTALL_MOD_STRIP=1 \
 		INSTALL_MOD_PATH=$(KERNEL_DIR_DRV_KO)
-	$(AT)find $(KERNEL_DIR_DRV_KO)/lib/modules/*/kernel \
-		-name "*.ko" -type f -exec cp -fav {} $(SYSDRV_KERNEL_MOD_PATH) \;
+	$(AT)cp -rfa $(KERNEL_DIR_DRV_KO)/lib $(SYSDRV_KERNEL_MOD_PATH)/
 	@test ! -d drv_ko || ($(MAKE) O=$(SYSDRV_KERNEL_OBJS_OUTPUT_DIR) -C $(SYSDRV_DIR)/drv_ko KERNEL_DRV_KO_INSTALL_PATH=$(SYSDRV_KERNEL_MOD_PATH))
+	$(AT)find $(SYSDRV_KERNEL_MOD_PATH) \( -name "build" -o -name "source" \) -exec rm -rf {} +
 	$(call MAROC_COPY_PKG_TO_SYSDRV_OUTPUT, $(OUTPUT_ROOTFS_SOURCE_DIR), $(SYSDRV_KERNEL_MOD_PATH), SELF)
 
 drv_clean:
